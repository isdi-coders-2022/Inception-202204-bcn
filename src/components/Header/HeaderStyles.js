import styled from "styled-components";

const HeaderStyles = styled.header`
  width: 100%;
  height: 150px;
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #11001c;
  color: #fff;
  padding: 20px;

  .info__header {
    margin: 0 auto;
  }

  img {
    width: 140px;
    height: 70px;
  }
  .seacher {
    display: flex;
    align-items: center;
    margin-top: 10px;
    input {
      border-radius: 40px;
      height: 30px;
      width: 50%;
      text-align: center;
      margin-left: 6%;
    }
  }
`;

export default HeaderStyles;
