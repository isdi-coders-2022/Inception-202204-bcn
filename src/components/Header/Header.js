import HeaderStyles from "./HeaderStyles";

const Header = () => {
  return (
    <HeaderStyles>
      <div className="info__header">
        <img src="/images/logo-supersimple.png" alt="logo" />
        <div className="seacher">
          <p>Search Cocktail:</p>
          <input title="serch-coctail" type="text" />
        </div>
      </div>
    </HeaderStyles>
  );
};

export default Header;
