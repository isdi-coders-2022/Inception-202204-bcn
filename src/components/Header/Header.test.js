import { render, screen } from "@testing-library/react";
import Header from "./Header";

describe("Given a Header component", () => {
  describe("When it's instantiated", () => {
    test("Then it should render a div with image inside", () => {
      const image = "/images/logo-supersimple.png";

      render(<Header />);

      const expectedImageRendered = screen.getByRole("img");

      expect(expectedImageRendered.src).toContain(image);
    });

    test("Then it shoul render a text 'Search Cocktail:'", () => {
      const text = "Search Cocktail:";

      render(<Header />);

      const exepectedTextRendered = screen.getByText(text);

      expect(exepectedTextRendered).not.toBeNull();
    });
  });
});
