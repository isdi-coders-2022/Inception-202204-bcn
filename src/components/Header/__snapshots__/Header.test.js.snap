// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Header component When it's instantiated Then it should always match this snapshot 1`] = `
<header
  className="sc-bczRLJ jJNOkx"
>
  <div
    className="info__header"
  >
    <img
      alt="Logotipo "
      src="/images/logo-supersimple.png"
    />
    <div
      className="seacher"
    >
      <p>
        Search Cocktail:
      </p>
      <input
        type="text"
      />
    </div>
  </div>
</header>
`;
